darshan-util/pydarshan/darshan/tests/test_plot_dxt_heatmap.py

```diff
@@ -115,21 +93,19 @@ def test_set_x_axis_ticks_and_labels(
     if filepath is None:
         # don't have any data sets with a max time between 1 and 10, so
         # create a synthetic one here
-        data = [[4, 1.03378843, 1.03387713, 0], [4000, 1.04216653, 1.04231459, 0]]
-        cols = ["length", "start_time", "end_time", "rank"]
-        agg_df = pd.DataFrame(data=data, columns=cols)
+        runtime = 1
```

Comment:
hmm, the comment above doesn't really make sense anymore, and the basis for deleting the previous test case is?

Reply:
Yeah, the old `tmax_dxt` was `1.04231459`, so by the new logic the closest thing to that old case would seem to be something like this:

```diff
@@ -93,8 +93,13 @@ def test_set_x_axis_ticks_and_labels(
     if filepath is None:
         # don't have any data sets with a max time between 1 and 10, so
         # create a synthetic one here
+        data = [[4, 1.03378843, 1.03387713, 0], [4000, 1.04216653, 1.04231459, 0]]
+        cols = ["length", "start_time", "end_time", "rank"]
+        agg_df = pd.DataFrame(data=data, columns=cols)
+        tmax_dxt = float(agg_df["end_time"].max())
         runtime = 1
-
+        if tmax_dxt > runtime:
+            runtime += 1
```

Not sure how you make that pass the test, but then this is the new logic/approach, as far as I can tell. I guess you'd need all the scaling stuff as well.

I guess my overall point is just that this testing change is a bit weird on its own, since it seems to merely allow the test to pass instead of representing the old case, which indeed appears to be a useful scenario to test (though I'm not sure things are factored correctly to test it here, or if much is gained by pasting the source of the DXT plotting code here as well).

Reply:
It is probably the case that the axis scaling logic is drifting a bit from a set of unified unit-testable function(s) because of the addition of all of this stuff outside of the actual function:

```python
    # scale the x-axis to span the calculated run time
    scale_factor = runtime / tmax_dxt
    jgrid.ax_joint.set_xlim(0.0, np.ceil(xbins * scale_factor))
    # rescale x-ticks so x-ticks are adjusted properly downstream
    jgrid.ax_joint.set_xticks(scale_factor * np.asarray(jgrid.ax_joint.get_xticks()))
```

I didn't check if things get cleaner to test if that gets kind of packed into `set_x_axis_ticks_and_labels()`, though it is possible with a func signature change I suspect.

Reply:
Iterating on this more, it seems moving the below code into `set_x_axis_ticks_and_labels()` makes the test a lot more difficult to manage. 

```python
    # rescale x-ticks so x-ticks are adjusted properly downstream
    jgrid.ax_joint.set_xticks(scale_factor * np.asarray(jgrid.ax_joint.get_xticks()))
```

Instead I think I could pass in a new formal parameter `bin_max` (`xbins * scale_factor`) and change `get_x_axis_ticks()` to use that value instead of pulling it from the axis object. The benefit to this is we no longer need the above line of code, and I think it makes the code simpler overall. The pasted diff below shows what this would look like in comparison. 

<details><summary>diff</summary>

```diff 
--- a/darshan-util/pydarshan/darshan/experimental/plots/plot_dxt_heatmap.py
+++ b/darshan-util/pydarshan/darshan/experimental/plots/plot_dxt_heatmap.py
@@ -28,7 +28,7 @@ import darshan
 from darshan.experimental.plots import heatmap_handling
 
 
-def get_x_axis_ticks(ax: Any, n_xlabels: int = 4) -> npt.NDArray[np.float64]:
+def get_x_axis_ticks(bin_max: float, n_xlabels: int = 4) -> npt.NDArray[np.float64]:
     """
     Creates the x-axis tick mark locations.
 
@@ -42,24 +42,10 @@ def get_x_axis_ticks(ax: Any, n_xlabels: int = 4) -> npt.NDArray[np.float64]:
 
     """
-    # get the original x-axis tick locations
-    initial_xticks = ax.get_xticks()
-    if len(initial_xticks) < n_xlabels:
-        # if there are too few initial x-axis tick locations, generate
-        # a new array with 0 as the minimum and the same maximum as
-        # the original x-axis ticks
-        xticks = np.linspace(0, np.max(initial_xticks), n_xlabels)
-    else:
-        # use the original tick marks to make new arrays that
-        # contain a subset of the original ticks/labels
-        tick_idx = np.round(np.linspace(0, initial_xticks.size - 1, n_xlabels)).astype(
-            int
-        )
-        xticks = initial_xticks[tick_idx]
-    return xticks
+    return np.linspace(0, bin_max, n_xlabels)
 
 
 def get_x_axis_tick_labels(
@@ -185,7 +171,7 @@ def get_y_axis_tick_labels(ax: Any, n_ylabels: int = 6) -> npt.NDArray[np.intc]:
 
 
 def set_x_axis_ticks_and_labels(
-    jointgrid: Any, tmax: float, n_xlabels: int = 4
+    jointgrid: Any, tmax: float, bin_max: float, n_xlabels: int = 4
 ):
     """
     Sets the x-axis tick mark locations and labels.
@@ -201,7 +187,7 @@ def set_x_axis_ticks_and_labels(
 
     """
     # retrieve the x-axis tick mark locations and labels
-    xticks = get_x_axis_ticks(ax=jointgrid.ax_joint, n_xlabels=n_xlabels)
+    xticks = get_x_axis_ticks(bin_max=bin_max, n_xlabels=n_xlabels)
     xticklabels = get_x_axis_tick_labels(max_time=tmax, n_xlabels=n_xlabels)
     # set the x-axis ticks and labels
     jointgrid.ax_joint.set_xticks(xticks)
@@ -404,13 +390,10 @@ def plot_heatmap(
     if tmax_dxt > runtime:
         runtime += 1
     # scale the x-axis to span the calculated run time
-    scale_factor = runtime / tmax_dxt
-    jgrid.ax_joint.set_xlim(0.0, np.ceil(xbins * scale_factor))
-    # rescale x-ticks so x-ticks are adjusted properly downstream
-    jgrid.ax_joint.set_xticks(scale_factor * np.asarray(jgrid.ax_joint.get_xticks()))
-
+    xbin_max = xbins * (runtime / tmax_dxt)
+    jgrid.ax_joint.set_xlim(0.0, xbin_max)
     # set the x and y tick locations and labels using the runtime
-    set_x_axis_ticks_and_labels(jointgrid=jgrid, tmax=runtime, n_xlabels=4)
+    set_x_axis_ticks_and_labels(jointgrid=jgrid, tmax=runtime, bin_max=xbin_max, n_xlabels=4)
     set_y_axis_ticks_and_labels(jointgrid=jgrid, n_ylabels=6)
````

</details>

If this seems "okay" I'll go ahead and fix the tests that this breaks, and push it in.